interview-questions.md

```diff
@@ -4,49 +4,74 @@ Answer the following questions. First, without external resources. Challenge you
 
 
 2. What is a virtual DOM?
 
-  Your answer:
+  Your answer: REACT has a virtual DOM which means it refreashes on its own without having to hard reload a whole page when a function is being done. For example when executing a function like an onClick function. The page does not need a hard reload. It will be updated automatically. 
+
+  Researched answer: The virtual DOM (VDOM) is a programming concept where an ideal, or “virtual”, representation of a UI is kept in memory and synced with the “real” DOM by a library such as ReactDOM. This process is called reconciliation.
 
-  Researched answer:
+  React creates a tree of custom objects representing a part of the DOM. For example, instead of creating an actual DIV element containing a UL element, it creates a React.div object that contains a React.ul object. It can manipulate these objects very quickly without actually touching the real DOM or going through the DOM API. Then, when it renders a component, it uses this virtual DOM to figure out what it needs to do with the real DOM to get the two trees to match.
+
+You can think of the virtual DOM like a blueprint. It contains all the details needed to construct the DOM, but because it doesn't require all the heavyweight parts that go into a real DOM, it can be created and changed much more easily.
 
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
-  Your answer:
+  Your answer: You need to first make a new directory called components. Inside components you make a new file with the name of the component. Make sure to follow the same React basic boilerplate with the name of your file in the appropriate places. Be sure to export the file. And make sure to import it into the App.js file. Inside the new file make the changes needed to diplay "Hello World". You can use a variety of different ways to write out the "Hello Word" (h1, div, p) however be sure it is in the new files return function. Inside the app.js file put <NewFile /> inside of the App.js return function. Once that is done you should see the contents of your new file on the browser after dragging the file into it
+
 
-  Researched answer:
+  Researched answer: React lets you define components as classes or functions. Components defined as classes currently provide more features which are described in detail on this page. To define a React component class, you need to extend React.Component
+  The only method you must define in a React.Component subclass is called render(). All the other methods described on this page are optional.
 
+  These methods are called in the following order when an instance of a component is being created and inserted into the DOM:
+constructor()
+static getDerivedStateFromProps()
+render()
+componentDidMount()
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JSX is the javascript in REACT. It allows us to write HTML in javascript
```

Comment:
Totally! I think it would be wise to break down the name of JSX as well, what it means.